client/src/paths/AssignmentViewer.tsx

```diff
@@ -153,6 +142,17 @@ export async function loader({ params }: { params: any }) {
     );
   }
 
+  // Due to the fact that the API `getAssignmentViewerDataFromCode` has the side effect
+  // of creating an anonymous user if not logged in, we wait until that happens and
+  // then redirect to the enter name screen.
+  // One downside of this is that we load the assignment data twice, once at
```

Comment:
If we are worried about the extra query, we could change `assign.ts/getAssignmentViewerDataFromCode` to first check if `lastNames` is blank and just shortcut return something like `{blankName: true}` in that case. I'd hold off on that, though, as it isn't a clear win given that it creates an extra user query in the case where you are logged in.